### PR TITLE
introduce a getSdkPath() API

### PR DESCRIPTION
**Pull request by devoncarew**
- introduce a `getSdkPath()` API
- some surgery to `getSdkDir` to use dart:io's Platform.resolvedExecutable API
- use package:test
- remove some now unused deps
- speculatively bump the version

@pq, comments appreciated!

### CODE REVIEW

File: lib/cli_util.dart [hunk 1]
```diff
@@ -6,11 +6,16 @@ library cli_util;
 
 import 'dart:io';
 
-import 'package:path/path.dart' as p;
-import 'package:which/which.dart';
```
Comment (pq) [inline lib/cli_util.dart]: 👍 Awesome to lose this dependency.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,12 +1,14 @@
 name: cli_util
-version: 0.0.1+3
+version: 0.1.0-dev.1
 author: Dart Team <misc@dartlang.org>
 description: A library to help in building Dart command-line apps.
 homepage: https://github.com/dart-lang/cli_util
+
 environment:
-  sdk: '>=1.0.0 <2.0.0'
+  sdk: '>=1.11.0 <2.0.0'
```
Comment (kevmoo) [inline pubspec.yaml]: Has this new API be around since `1.11`?

Comment (devoncarew) [inline pubspec.yaml]: Yup, at least from my brief perusal of the changelog.

**Comment by kevmoo** [comment]
@devoncarew plans to publish this soon? It'd clean up a lot of uses of when, I'm sure.

File: pubspec.yaml (hunk 2 above)
Comment (kevmoo) [inline pubspec.yaml]: Indeed!

**Comment by kevmoo** [comment]
Wow...I'd love to see what happens to usages of `when` and `which` when this lands. Would be great to clean up.

**Comment by devoncarew** [comment]
I was thinking of adding some basic console ux / progress output, but not clear on the timeframe for that. We can go from `0.1.0-dev.1` to `0.1.0` and publish now -

**Comment by kevmoo** [comment]
That's a PR I'd happily review. 😄

**Pull request merged**